Waybar/weather

Guillaumecoi opened the pull request:
# Pull Request

Request is based on [issue 252](https://github.com/HyDE-Project/HyDE/issues/252)

## Description

Tweaks to the weather Script to make it more versatile

- Weather in waybar now has different configurations that can be set in `~/.local/state/hyde/config`.
- Variable that can be set, the options, default:
    - TEMP_UNIT :  c or f (default: c)
    - TIME_FORMAT: 12h or 24h (default: 12h)
    - WINDSPEED_UNIT: km/h or mph (default: Km/h)
    - SHOW_ICON: True or False (default: True)
    - SHOW_LOCATION: True or False (default: False)
    - SHOW_TODAY_DETAILS: True or False (default: True)
    - FORECAST_DAYS: 0,1,2,3 (default: 3)
    - WAYBAR_WEATHER_LOC: Placename (default: '', based on IP location)
    If the input would be invalid it is automatically changed into the default.
- The standard layout has been made easier to read.

## Type of change

Please put an `x` in the boxes that apply:

- [ x] **Bug fix** (non-breaking change which fixes an issue)
- [ x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Documentation update** (non-breaking change; modified files are limited to the documentations)
- [ ] **Technical debt** (a code change that does not fix a bug or add a feature but makes something clearer for devs)
- [ ] **Other** (provide details below)

## Checklist

Please put an `x` in the boxes that apply:

- [ x] I have read the [CONTRIBUTING](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md) document.
- [ x] My code follows the code style of this project.
- [ x] My commit message follows the [commit guidelines](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added a changelog entry.
- [ x] I have added necessary comments/documentation to my code.
- [ x] I have added tests to cover my changes.
- [ x] I have tested my code locally and it works as expected.
- [ x] All new and existing tests passed.

## Screenshots
How it it was: (ignore my clock)

![image](https://github.com/user-attachments/assets/ef79207a-1ee0-4fb4-a8c8-d3048fb3c41e)

How it is now with default settings:

![image](https://github.com/user-attachments/assets/a73cc420-c6a9-4aa1-a1ec-d9a770faec0e)

Example with changed settings:

![image](https://github.com/user-attachments/assets/52251fb0-c5a9-4640-9a2e-96ec003df329)
![image](https://github.com/user-attachments/assets/4fff96dd-0184-4794-b98c-81fd9d98e433)


## Additional context
- I could write a script so that clock also uses the variable to display 12h or 24h
- Extra settings are possible as long as it can be retrieved from this api: https://wttr.in/?format=j1

kRHYME7 commented:
Thanks, @Guillaumecoi. In this repo, we exposed `~/.config/hyde/config.toml`. 

https://github.com/HyDE-Project/HyDE/blob/ec19995601392c6660a5cf64d56f6b8b86eec0b5/Configs/.local/share/hyde/config.toml#L129C1-L131C1
example: 
```
[waybar.weather]
loc= true
```

As bash do not have an ability to parse a toml file, we have an intermediary script that watches the `~/.config/hyde/config.toml` for changes. Once changed, it will generate the state file `~/.local/state/hyde/config `. 

So I would suggest renaming the variables with `WEATHER_` prefixes.


```

    WEATHER_TEMP_UNIT : c or f (default: c)
    WEATHER_TIME_FORMAT: 12h or 24h (default: 12h)
    WEATHER_WINDSPEED_UNIT: km/h or mph (default: Km/h)
    WEATHER_SHOW_ICON: True or False (default: True)
    WEATHER_SHOW_LOCATION: True or False (default: False)
    WEATHER_SHOW_TODAY_DETAILS: True or False (default: True)
    WEATHER_FORECAST_DAYS: 0,1,2,3 (default: 3)
```
   

For this one let's suport 2 variables for now, and will deprecate soon. 
```
WAYBAR_WEATHER_LOC: Placename (default: '', based on IP location)
WEATHER_LOC: Placename (default: '', based on IP location)
```
Also please note that `WAYBAR_WEATHER_LOC` is also used to disable the location. Not all wants there location to be shown in a screenshot.

Guillaumecoi commented:
Thank you for your explanation since I did not get how the `.toml` file worked. I really like this solution to handle env variables, it is very clean.

- I changed the variable names in the Python script and tested it by setting the variables in `~/.config/hyde/config.toml`.

- I also set the default variables in `~.local/share/hyde/config.toml`. So it is more clear how to use them.

#### I still have some questions:
- "For this one let's support 2 variables for now, and will deprecate soon." My Script does not read the variable WAYBAR_WEATHER_LOC anymore and that is why I deleted it. Is it best to avoid this so people don't lose there loc setting upon updating? If so wouldn't it be better to place all variables under: `[waybar.weather]` so that I dont have to check 2 variables?

- Is it correct that the variables in `~/.config/hyde/config.toml` are not used but are more like a template to show how you can use `~/.config/hyde/config.toml`. I think from taking a look at the scripts it does not use these values but it uses the default values in the script which are identical to those in `~/.config/hyde/config.toml`

dieBakterie reviewed (dismissed):
Fix the typos in [config.toml](https://github.com/HyDE-Project/HyDE/pull/254/files#diff-23f0f1bc0adbf87e1ed180527836cea664c4a551666e6baa48336e0b6b8f8dbe)

kRHYME7 commented:
@Guillaumecoi  

![image](https://github.com/user-attachments/assets/c15af7d9-fca5-4200-b327-6941cbe2f892)


1. Toml File - This is for users main interface
2. Translator - Intermediary script to trnslate the toml file into env variables
3. Env variables - These are variables  for bash/shell scripts to use. 
---

I hope this is pretty clear now. 

We knew that python has a toml lib but for now I want to unify the parsers in a single module so I want this MR to use the environemnt variable for now. 


> If so wouldn't it be better to place all variables under: [waybar.weather] so that I dont have to check 2 variables?

Let's break it. I guess only one user is using the feature as of the moment, I'll try to reach out to him  on merge.

kRHYME7 commented:
Let's just unify it under  [weather].

kRHYME7 commented:
@Guillaumecoi  Ready for merge?

kRHYME7 reviewed (approved):
Thank you @Guillaumecoi

Guillaumecoi commented:
I was looking for the typos but it seems I was a few minutes too late. For me it can be merged. And thank you for the egrapth

kRHYME7 commented:
@0xGeN02  Here's the config change. Thank you 

```
[weather]
show_location=true
location = ''			# Location/coordinates string for the weather output
```
